packages/sdk-core/src/modules/storage/SessionFiles.ts

```diff
@@ -0,0 +1,244 @@
+import { Events } from '../../common/Events';
+import { IdGenerator } from '../../common/IdGenerator';
+import { BacktraceModule } from '../BacktraceModule';
+import { FileSystem } from './FileSystem';
+
+interface FileSession {
+    readonly file: string;
+    readonly sessionId: string;
+    readonly escapedSessionId: string;
+    readonly timestamp: number;
+}
+
+type SessionEvents = {
+    unlocked(): void;
+};
+
+const SESSION_MARKER_PREFIX = 'bt-session';
+
+const isDefined = <T>(t: T | undefined): t is T => !!t;
+
+export class SessionFiles implements BacktraceModule {
+    public readonly marker: string;
+
+    private readonly _timestamp: number;
+    private readonly _events = new Events<SessionEvents>();
+    private readonly _escapedSessionId: string;
+    private readonly _locks = new Set<string>();
+    private _previousSession?: SessionFiles;
+    private _cleared = false;
+
+    constructor(
+        private readonly _fileSystem: FileSystem,
+        private readonly _directory: string,
+        public readonly sessionId: string,
+        private readonly _maxPreviousLockedSessions = 1,
+        timestamp?: number,
+        private readonly _lockable = true,
+    ) {
+        this._timestamp = timestamp ?? Date.now();
+        this._escapedSessionId = SessionFiles.escapeFileName(sessionId);
```

Comment:
why you prefer to use static instead of class member function?

Reply:
Previously they were used in a static scope. Removed `static`s.